packages/common/src/ContractInteractor.ts

```diff
@@ -462,20 +445,20 @@ export class ContractInteractor {
     return this.relayCallMethod(_.maxAcceptanceBudget, _.relayRequest, _.signature, _.approvalData).encodeABI()
   }
 
-  async getPastEventsForHub (extraTopics: string[], options: PastEventOptions, names: EventName[] = ActiveManagerEvents): Promise<EventData[]> {
+  async getPastEventsForHub (extraTopics: string[][], options: PastEventOptions, names: EventName[] = ActiveManagerEvents): Promise<EventData[]> {
```

Comment:
I wanted to check that. So you are saying, if I have an event and I want to search by the 3rd indexed field, I need to do `topics = [undefined, undefined, [value3]]`?
And that `[[], [], [value3]]` will not work?

As for "explicit topic" I think it is equivalent to the one-element array so I don't want to add it to the type, is that ok?

Reply:
I'm not saying your format will not work - it will - its just verbose, and harder to read.
all the following are equivalent  for the rpc APIs, and are also valid for the suggested types. it will also remove the need to modify all the places in the code that are otherwise unchanged.

`[null,null,topic3]`
`[[],null,topic3]`
`[[][],[topic3]]`
`[null,null,[topic3]`